bundle-dependencies.js

```diff
@@ -0,0 +1,121 @@
+const bundling = new Map()
+let recording = true
+
+// Note that append-transform and its dependencies cannot be bundled.
+const appendTransform = require('append-transform') // eslint-disable-line
+appendTransform((code, filename) => {
+  if (recording) {
+    bundling.set(filename, code)
+  }
+
+  return code
+})
+
+require(process.argv[2])
+recording = false
+
+// Load further dependencies after recording the files that are to be bundled.
+const Module = require('module')
+const { dirname } = require('path')
+const { transform } = require('babel-core')
+const wrapListener = require('babel-plugin-detective/wrap-listener')
+const resolveFrom = require('resolve-from')
+
+const rewriteRequire = wrapListener((path, file) => {
```

Comment:
I wonder if just using [`detective`](https://www.npmjs.com/package/detective) would be more performant than loading up all of babel. `babel-detective` is beneficial when we are using babel to perform other transforms, not sure how it compares performance-wise to the original.